JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -8,169 +17,189 @@
 import java.util.LinkedList;
 import java.util.Map;
 import java.util.HashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
-import ru.javafiddle.jpa.entity.Access;
-import ru.javafiddle.jpa.entity.UserGroup;
-import ru.javafiddle.jpa.entity.Group;
-import ru.javafiddle.jpa.entity.User;
 
 
-@Stateless
 
+@Stateless
 public class GroupBean {
 
-    @PersistenceContext
+    @EJB
+    private UserGroupBean userGroup;
+
+    private static final Logger logger =
+            Logger.getLogger(ProjectBean.class.getName());
+
+    @PersistenceContext(name = "JFPersistenceUnit")
     EntityManager em;
 
     public GroupBean() {
+
     }
 
-    //we should get at first userId, then add this element to UserGroups
-    //when we are in this method we definitly have group with such id
-    public void addMember(int groupId, String userNickName, String accessRights) {
-
-        //check if this group exists
-       /* Group group;
-        if (groupId == 0) {
-            group = new Group();
-            group.setGroupName("default");//still it is not clear about group names
-            em.persist(group);
-        }*/
-        User user = getUser(userNickName);
-        Access access = getAccess(accessRights);
+    //we always know that the group were we are trying to add a user exits now
+    //CALL THIS METHOD WHEN THE PERSON IS CREATING A GROUP BY HIMSELF!!
+    @TransactionAttribute(TransactionAttributeType.REQUIRES_NEW)
```

Comment:
REQUIRES_NEW is a dangerous thing, very rarely necessary. Please see my comment regarding to that.